Assets/Scripts/Data Model/AbstractLayer.cs

```diff
@@ -1,406 +1,408 @@
+
+			if (layerMeta.layer_editing_type == "protection")
 			{
-				layerstateObject = JsonConvert.DeserializeObject<List<LayerStateObject>>(layerMeta.layer_states);
+				MultiTypeSelect = true;
+				LayerManager.protectedAreaLayers.Add(this);
 			}
-			catch
+			else
 			{
-				Debug.LogError("Failed to deserialize: " + FileName + " trying to parse " + layerMeta.layer_states);
+				MultiTypeSelect = layerMeta.layer_editing_type == "multitype";
 			}
 
-			foreach (LayerStateObject obj in layerstateObject)
+			Selectable = layerMeta.layer_selectable;
+			Editable = layerMeta.layer_editable;
+			Toggleable = layerMeta.layer_toggleable;
+			ActiveOnStart = layerMeta.layer_active_on_start || Main.LayerVisibleForCurrentExpertise(layerMeta.layer_name);
+			greenEnergy = layerMeta.layer_green == 1;
+			Optimized = !Selectable && !Editable && !Toggleable && ActiveOnStart;
+
+			ParseEntityPropertyMetaData(layerMeta.layer_info_properties);
+			EntityTypes = ParseEntityTypes(layerMeta.layer_type);
+			visibleEntityTypes.UnionWith(EntityTypes.Values);
+			if(layerMeta.layer_text_info != null && layerMeta.layer_text_info.property_per_state != null)
+				textInfo = new LayerTextInfo(layerMeta.layer_text_info);	
+		
+			if (layerMeta.layer_states != "")
 			{
-				layerStates.Add(obj.state, obj.time);
+				List<LayerStateObject> layerstateObject = new List<LayerStateObject>();
+				try
+				{
+					layerstateObject = JsonConvert.DeserializeObject<List<LayerStateObject>>(layerMeta.layer_states);
+				}
+				catch
+				{
+					Debug.LogError("Failed to deserialize: " + FileName + " trying to parse " + layerMeta.layer_states);
+				}
+
+				foreach (LayerStateObject obj in layerstateObject)
+				{
+					layerStates.Add(obj.state, obj.time);
+				}
+
+				if (layerStates.ContainsKey(ASSEMBLY_STATE))
+					AssemblyTime = layerStates[ASSEMBLY_STATE];
 			}
 
-			if (layerStates.ContainsKey(ASSEMBLY_STATE))
-				AssemblyTime = layerStates[ASSEMBLY_STATE];
+			switch (layerMeta.layer_editing_type)
+			{
+				case "cable":
+					editingType = EditingType.Cable;
+					break;
+				case "transformer":
+					editingType = EditingType.Transformer;
+					LayerManager.AddEnergyPointLayer(this as PointLayer);
+					break;
+				case "socket":
+					editingType = EditingType.Socket;
+					LayerManager.AddEnergyPointLayer(this as PointLayer);
+					break;
+				case "sourcepoint":
+					editingType = EditingType.SourcePoint;
+					LayerManager.AddEnergyPointLayer(this as PointLayer);
+					break;
+				case "sourcepolygon":
+					editingType = EditingType.SourcePolygon;
+					break;
+				default:
+					editingType = EditingType.Normal;
+					break;
+			}
+			if (editingType != EditingType.Normal)
+			{
+				presetProperties.Add("MaxCapacity", (subent) => 
+				{
+					ValueConversionCollection valueConversions = VisualizationUtil.VisualizationSettings.ValueConversions;
+					IEnergyDataHolder data = (IEnergyDataHolder)subent;
+					//return data.Capacity.ToString();
```

Comment:
Commented code, as always see if it makes sense to leave it there for the future, otherwise remove it! :)

Also lines 168, 172, 183